overlord/snapstate/snapstate.go

```diff
@@ -2431,6 +2438,87 @@ func checkDiskSpace(st *state.State, changeKind string, infos []minimalInstallIn
 	return nil
 }
 
+// MigrateHome migrates a set of snaps to use a ~/Snap sub-directory as HOME.
+// The state must be locked by the caller.
+func MigrateHome(st *state.State, snaps []string) ([]*state.TaskSet, error) {
+	tr := config.NewTransaction(st)
+	moveDir, err := features.Flag(tr, features.MoveSnapHomeDir)
+	if err != nil {
+		return nil, err
+	}
+
+	if !moveDir {
+		_, confName := features.MoveSnapHomeDir.ConfigOption()
+		return nil, fmt.Errorf("cannot migrate to ~/Snap: flag %q is not set", confName)
```

Comment:
Does this mean the user needs to set the "flag" manually when doing a `snap debug move-homedirs` ? Or is that set implicitly when doing the snap debug command? And if implicit do we actually need it given that it needs to be triggered explitly

Reply:
The flag needs to be set separately before. That's intentional and correct and as discussed with Gustavo afair. The reason is that there is no clean automatic undo for this, so one needs to express somehow to be OK with the risks.